src/main/java/randoop/generation/Impurity.java

```diff
@@ -0,0 +1,518 @@
+   * @param outputClass the class of the primitive number to be fuzzed
+   * @return a sequence with the fuzzing statement appended at the end
+   */
+  private static Sequence getFuzzedSequenceForPrimNumber(Sequence sequence, Class<?> outputClass) {
+    Object fuzzedValue = getFuzzedValueForPrim(outputClass);
+    Sequence fuzzingSequence = Sequence.createSequenceForPrimitive(fuzzedValue);
+    List<Sequence> temp = new ArrayList<>(Collections.singletonList(sequence));
+    temp.add(fuzzingSequence); // Add fuzzing sequence to the list
+    return Sequence.concatenate(temp); // Assuming concatenate combines all sequences in the list
+  }
+
+  /**
+   * Get a fuzzed value for a primitive number using a Gaussian distribution.
+   *
+   * @param outputClass the class of the primitive number to be fuzzed
+   * @return a fuzzed value for the primitive number
+   */
+  private static Object getFuzzedValueForPrim(Class<?> outputClass) {
+    double randomGaussian = GAUSSIAN_STD * Randomness.nextRandomGaussian(1);
+    if (outputClass == int.class) {
+      return (int) Math.round(randomGaussian);
+    } else if (outputClass == short.class || outputClass == byte.class) {
+      return (short) Math.round(randomGaussian); // Unified handling for short and byte
+    } else if (outputClass == long.class) {
+      return Math.round(randomGaussian);
+    } else if (outputClass == float.class) {
+      return (float) randomGaussian;
+    } else if (outputClass == double.class) {
+      return randomGaussian;
+    } else {
+      throw new RuntimeException("Unexpected primitive type: " + outputClass.getName());
+    }
+  }
+
+  /**
+   * Get the method (in a list) that can be used to fuzz primitive numbers.
+   *
+   * @param outputClass the class of the primitive number to be fuzzed
+   * @return a list of methods that can be used to fuzz primitive numbers
+   * @throws NoSuchMethodException if no suitable method is found for the given class
+   */
+  private static List<Method> getNumberFuzzingMethod(Class<?> outputClass)
+      throws NoSuchMethodException {
+
+    List<Method> methodList = new ArrayList<>();
+
+    // Map each wrapper to its primitive type and a common method
+    if (outputClass == int.class) {
+      methodList.add(Integer.class.getMethod("sum", int.class, int.class));
+    } else if (outputClass == double.class) {
+      methodList.add(Double.class.getMethod("sum", double.class, double.class));
+    } else if (outputClass == float.class) {
+      methodList.add(Float.class.getMethod("sum", float.class, float.class));
+    } else if (outputClass == long.class) {
+      methodList.add(Long.class.getMethod("sum", long.class, long.class));
+    } else if (outputClass == short.class) {
+      methodList.add(Integer.class.getMethod("sum", int.class, int.class));
+    } else if (outputClass == byte.class) {
+      throw new NoSuchMethodException("Byte fuzzing is not supported yet");
```

Comment:
`NoSuchMethodException` is intended for problems with dynamic dispatch, not for situations like this.  Use `RandoopBug` or `Error` or the like.